ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,407 @@
+"""
+Define a parent IrfTool class to hold all the options
+"""
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable
+from pyirf.binning import create_bins_per_decade
+
+from ..core import Component, QualityQuery, traits
+from ..core.traits import Bool, Float, Integer, List, Unicode
+
+
+class ToolConfig(Component):
```

Comment:
Tools can have their own config. There is now point in introducing a `Component` just to hold configuration for a tool. This also makes config files awkward, since you would have to write:

```yaml
IRFTool:
    ToolConfig:
        gamma_file: "path/to/gamma_file"
```

Instead of just:
```yaml
IRFTool:
    gamma_file: "path/to/gamma_file"
```

Reply:
implemented